### PR TITLE
Allow an empty array of repo ids as a request parameter

### PR DESCRIPTION
**Pull request by Matthew-Reidy**
Fixes: #3106.

Add extra endpoint function that allows client to pass an empty array of repository ids to the installations/<install_id>/access_tokens endpoint as a parameter. Response lists all repositories in an installation.

### CODE REVIEW

File: github/apps.go [hunk 1]
```diff
@@ -344,6 +358,28 @@ func (s *AppsService) CreateInstallationToken(ctx context.Context, id int64, opt
 	return t, resp, nil
 }
 
+// CreateInstallationTokenListRepos creates a new installation token with an empty repository list as a parameter.
```
Comment (gmlewis) [inline github/apps.go:361]: I'm thinking that we should mention _why_ this method exists and what makes it different from the other one.

File: github/apps.go [hunk 2]
```diff
@@ -344,6 +358,30 @@ func (s *AppsService) CreateInstallationToken(ctx context.Context, id int64, opt
 	return t, resp, nil
 }
 
+// CreateInstallationTokenListRepos creates a new installation token with a list of all repositories in an installation which is not possible with CreateInstallationToken.
+//
+// differs from CreateInstallationToken by taking InstallationTokenListRepoOptions as a parameter which does not omit RepositoryIDs if that field is nil or an empty array.
```
Comment (gmlewis) [inline github/apps.go:363]: nit: "differs" => "It differs"

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Matthew-Reidy !
One tiny nit so we have complete sentences in our godocs, otherwise LGTM.

File: github/apps.go (hunk 2 above)
Comment (Matthew-Reidy) [reply github/apps.go]: done

**Review by gmlewis — APPROVED** [review]
Thank you, @Matthew-Reidy !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by be0x74a — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @be0x74a !
Merging.

**Pull request merged**